### PR TITLE
escape commas only once when uploading an image with text overlay sin…

### PR DESCRIPTION
**Pull request by eitanp461**
…ce it is an eager transformation

### CODE REVIEW

File: src/utils.coffee [hunk 1]
```diff
@@ -86,7 +86,7 @@ process_if = (ifValue)->
 # Parse layer options
 # @return [string] layer transformation string
 # @private
-process_layer = (layer)->
+process_layer = (layer, escapeQuotes = true)->
```
Comment (shedar) [inline src/utils.coffee]: Can you suggest a better param name than `escapeQuotes`?
As I see, it's about comma and slash, not about quotes.

File: src/utils.coffee [hunk 2]
```diff
@@ -271,7 +273,7 @@ exports.generate_transformation_string = (options) ->
   if options["offset"]?
     [options["start_offset"], options["end_offset"]] = split_range(utils.option_consume(options, "offset"))
 
-  overlay = process_layer(utils.option_consume(options, "overlay"))
+  overlay = process_layer(utils.option_consume(options, "overlay"), escapeQuotes)
```
Comment (shedar) [inline src/utils.coffee]: Shouldn't it be added to the underlay as well?

File: test/utils_spec.coffee [hunk 3]
```diff
@@ -385,6 +385,18 @@ describe "utils", ->
...
+        overlay: {
+          text:"Hello,World", 
+          font_family:"arial", 
+          font_size: '30'   
+        }
+    }
+    expect(utils.build_upload_params(options)['transformation']).to.eql("l_text:arial_30:Hello%2CWorld")
+    expect(Object.keys(options).length).to.eql(1)
+
+
```
Comment (shedar) [inline test/utils_spec.coffee]: Is test of double escaping outside of `build_upload_params` intentionally omited to be tested in core or just missing?

File: src/utils.coffee (hunk 1 above)
Comment (asisayag2) [reply src/utils.coffee]: @shedar , how about "shouldEscape"?

File: src/utils.coffee (hunk 2 above)
Comment (asisayag2) [reply src/utils.coffee]: @shedar , we should

File: test/utils_spec.coffee (hunk 3 above)
Comment (asisayag2) [reply test/utils_spec.coffee]: @shedar missing

**Comment by shedar** [comment]
Replaced by #249

**Pull request closed without merging**